python/tank/authentication/core_defaults_manager.py

```diff
@@ -78,11 +77,15 @@ def get_user_credentials(self):
                   User or None in case no credentials could be established.
         """
         if not self._mask_script_user:
-            from . import shotgun
             data = shotgun.get_associated_sg_config_data()
             if data.get("api_script") and data.get("api_key"):
                 return {
                     "api_script": data["api_script"],
                     "api_key": data["api_key"]
                 }
         return super(CoreDefaultsManager, self).get_user_credentials()
+
+
+# For backwards compatibility.
```

Comment:
The `CoreDefaultsManager` used to be in `util` because it used the `shotgun_authentication` module which was considered outside of Toolkit. When we moved the authentication module inside core however, it introduced a circular dependency between util and the authentication module, as `util` is meant to be our module with the least dependencies on other modules.

Since the CoreDefaultsManager authentication related, it makes sense to move it there. Doing so finally breaks that circular dependency and allows to fix a couple of nasty local imports and make them global in the `sgtk.authentication.DefaultsManager` class for example. The code is finally cleaned up.

Now, when `core_detaults_manager.py` is imported at init time, it sets the util module so the code can remain backwards compatible.